KSF-30_RouterVue

Madridka opened the pull request.

Madridka committed:
add button hello Helloworld.vue

Madridka committed:
-added some styles in helloworld.vue
-delete button hello world
-delete button full reset

Madridka committed:
delete style in app.vue

Madridka committed:
feat: by default all inputs empty, title = "Добавление товара", button "добавить товар" not available (when inputs is empty) (p.3 KSF-16)

feat: when editing form, title = "Редактирование товара" (p.6 KSF-16)

Madridka committed:
add: styles.scss
add: btn-styles.scss
ref: all classes of buton (work with sass-css)

Madridka committed:
refactor: folder styles moves to assets from components
add: normalize.css
refactor: styles.scss change by rules of sass

Madridka committed:
refactor: rename HelloWorld to onlineShop
style: some changes in btn-styles.scss
refactor: move input editPrice
ref

Madridka committed:
refactor: move everything in to app.vue
add: added file addForm for input new product

Madridka committed:
fix: addForm.vue fixed, all props are working

this is the last update of branch KSF-16, next branch called KSF-23

Madridka committed:
feat: add textarea with description

fix: add editing textarea description

fix: change some style

Madridka committed:
feat: create modal window for add product (addForm.vue)

feat: add button to top to add new product and open modal

feat: create modal window to edit product (editForm)

style: add styles to modal window

style: add style to btn-close modal window

Madridka committed:
fix: delete TotalPrice from computed

Madridka committed:
refactor: delete formTitle from computed

Madridka committed:
fix: move disable from computed app.vue to addform.vue computed

Madridka committed:
feat: now you cant save your edit, whem form is empty (KSF-30: 7.

Madridka committed:
feat: add vue-router

Madridka committed:
feat: add 3 rows 4 columns grid boxes, each product card 200x300px

feat: add 12 product card on view

Madridka committed:
feat: add adminPage (infoPage delete)

Madridka committed:
feat: add store (vuex)
feat: move order from all components to store.js
feat: view order from store to homepage and adminpage

Madridka committed:
refactor: rename shopList to productList
refactor: rename all variables order to product
refactor: nav link from footer to header (temporarily)
refactor: create folder pages to homePage and adminPage
refactor: remove all excess methods, computed, props and other from vue components (new architecture)
refactor: change all methods (edit, save) to work with vuex

Madridka committed:
feat: add /product/:id to router.js
feat: add router-link to all of products on home page
style: remove all link style (home page -> router-link)

feat: add to store.js (and to all comp) new variables - fullDesc
feat: change variable description from store.js to shortDesc

style: font size btn-add change to 25px

Madridka committed:
style: add class "product" to each product in productPage.vue
style: class product from productList.vue change to products

feat: work with pictures (add to store to each product pic
feat: in admin you can swap pic to each product

feat: add page to each product (productPage.vue)

Madridka committed:
feat: in addForm.vue can add new product with pic upload

Madridka committed:
feat: add /cart to router.js
feat: product add to cart, when click Купить on productList.vue
feat: add cartPage (modal)

Madridka committed:
feat: in cartPage.vue realize full cart design with price, quantity, and full price of shopping

Madridka committed:
feat: + and - to quantity of card (if 0 card will be delete)

Madridka committed:
feat: add notification (with transition) when product added to cart

Madridka committed:
feat: add div Full Price of cart

Madridka committed:
feat: if === 0, full price is not visible

liberian commented on src/App.vue:
```diff
@@ -1,28 +1,40 @@
+          <router-link class="nav-link" exact to="/admin" active-class="active">Админка</router-link>
+        </li>
+        <li>
+          <router-link class="nav-link" exact to="/cart" active-class="active">Корзина</router-link>
```
будет лучше, если ссылки вывести в цикле

Madridka replied:
добавил вывод ссылок через цикл

liberian commented on src/pages/adminPage.vue:
```diff
@@ -0,0 +1,161 @@
+      this.editPrice = product.price;
+      this.editshortDesc = product.shortDesc;
+      this.editfullDesc = product.fullDesc;
+    },
```
здесь можно обойтись одной переменной product, это же объект. только его зарегать нужно в data и использовать при  редактировании, внутри же все свойства нужные лежат. Уйдут лишние 7 переменных, зачем память засорять)

Madridka replied:
завернул все лишние переменные в объект editProduct: null,

liberian commented on src/pages/adminPage.vue:
```diff
@@ -0,0 +1,161 @@
+        this.editPrice !== "" &&
+        this.editshortDesc !== "" &&
+        this.editfullDesc !== ""
+      );
```
после объявления переменной product в data здесь можно пройтись по ключам объекта и проверить везде ли есть значения. Плюс вынести этот метод в computed, но не забыть про реактивность свойств объекта, вспомни как она работает.

Madridka replied:
переделал, теперь идет обращение к объекту editProduct
см. последний commit

liberian commented on src/pages/adminPage.vue:
```diff
@@ -0,0 +1,161 @@
+        shortDesc: this.editshortDesc,
+        fullDesc: this.editfullDesc,
+      };
+      this.$store.commit("updateProduct", updatedProduct); 
```
здесь так же потом воспользуйся переменной product , если корректно настроишь реактивность, то в коммит уйдет измененный объект

Madridka replied:
добавил новый объект newProduct: {...} в addForm.vue
см. последний commit

liberian commented on src/pages/adminPage.vue:
```diff
@@ -0,0 +1,161 @@
+      this.editPrice = "";
+      this.editshortDesc = "";
+      this.editfullDesc = "";
+    },
```
здесь все так же , только с product взаимодействовать

Madridka replied:
сделано

liberian commented on src/pages/adminPage.vue:
```diff
@@ -0,0 +1,161 @@
+      this.$store.state.products = this.$store.state.products.filter(
+        (p) => p.id !== id
+      );
+    },
```
т.к. работаешь с сущностями стора, то лучше этот метод убрать в стор и через коммит его вызывать, не забыть про ИД

liberian commented on src/pages/cartPage.vue:
```diff
@@ -0,0 +1,70 @@
+      <h3>Корзина</h3>
+      <hr />
+      <ul>
+        <h3 v-if="$store.state.cart.length === 0">Корзина пуста</h3>
```
`$store.state.cart.length === 0"`  убрать в computed, сюда вернуть только результат. Сам тег лучше переместить перед списком <ul>, по условию показываешь или заглушку, или список товаров

Madridka replied:
исправил
вычисление перенес в computed, оставил только isEmpty
```
      <h3 v-if="isEmpty">Корзина пуста</h3>
      <h3 v-else>Корзина</h3>
```

liberian commented on src/pages/cartPage.vue:
```diff
@@ -0,0 +1,70 @@
+      <hr />
+      <ul>
+        <h3 v-if="$store.state.cart.length === 0">Корзина пуста</h3>
+        <li v-else v-for="card in $store.state.cart" :key="card.id">
```
v-else или v-if с v-for не желательно в одном теге использовать, как вариант обернуть в <template> и туда повесить условие отображения

Madridka replied:
замечание учту на будущее

liberian commented on src/pages/cartPage.vue:
```diff
@@ -0,0 +1,70 @@
+          <button class="btn btn-cancel" @click="deleteCard(card.id)">
+            удалить товар
+          </button>
+        </li>
```
в <li> содержится некий шаблон который можно вынести в отдельный компонент, со всеми вытекающими - пропсы и эмиты

liberian commented on src/pages/cartPage.vue:
```diff
@@ -0,0 +1,70 @@
+        </li>
+      </ul>
+      <div v-if="totalPrice === 0"></div>
+      <div v-else class="total">Полная стоимость: {{ totalPrice }} руб. </div>
```
здесь достаточно показать блок с итоговой ценой при totalPrice > 0

Madridka replied:
```
<div v-if="totalPrice > 0" class="total">
     Полная стоимость: {{ totalPrice }} руб.
</div>
```
исправил

liberian commented on src/pages/cartPage.vue:
```diff
@@ -0,0 +1,70 @@
+      return card.price * card.quantity;
+    },
+    increaseQuantity(card) {
+      this.$store.commit("increaseQuantity", card);
```
тут и ниже в методах наверное все же ИД передается, а то кажется что ты объект передаешь

Madridka replied:
разобрался, переделал, да
теперь везде идет обращение не к объекту card, а через id

Madridka committed:
fix: the order of options in all components (props-data-comp-watch...)
feat: use v-for to nav links
feat: in addForm.vue add new object "newProduct" (delete separate variables)
feat: del editForm (extra component)
feat: add object "editProduct" to adminPage.vue, delete separate variables
feat: move isEmpty "Корзина пуста" higher, ...length===0 move to computed
fix: delete div with totalPrice === 0, here stand totalPrice>0 now
fix: for methods in cartPage.vue increase/decrease change object to id

Madridka committed:
style: move import style to main.js
style: btn-styles.scss delete and move styles to baseButton
feat: add comp baseButton.vue, in other comp use this comp

Madridka committed:
feat: add component cartForm.vue

Madridka committed:
fix: error with isModalOpen in addForm.vue (move submit from form to baseButton, no need more var to modal window)
feat: add alert in addNewProduct, if new product is already used (in products: [])
feat: change props of additional class type in baseButton from "type" to "purpose"

Madridka committed:
feat: limitation count of product in cart (from homePage and cartPage)

Madridka committed:
fix: delete all console.log

Madridka committed:
feat: do not repeat name of product
